air-quality-ui/system_tests/ui/city_page/city_page.spec.ts

```diff
@@ -519,3 +522,180 @@ test.describe('Charts are fully visible in 1920x1080 viewport', () => {
     })
   })
 })
+
+test.describe('PM2.5 influences AQI plot', () => {
+  test.beforeEach(async ({ page, cityPage }) => {
+    const mockedForecastResponse = [
+      createForecastAPIResponseData({
+        base_time: '2024-07-01T00:00:00Z',
+        valid_time: '2024-07-01T00:00:00Z',
+        location_name: 'Rio de Janeiro',
+        overall_aqi_level: 1,
+      }),
+      createForecastAPIResponseData({
+        base_time: '2024-07-01T00:00:00Z',
+        valid_time: '2024-07-01T03:00:00Z',
+        location_name: 'Rio de Janeiro',
+        overall_aqi_level: 2,
+      }),
+      createForecastAPIResponseData({
+        base_time: '2024-07-01T00:00:00Z',
+        valid_time: '2024-07-01T06:00:00Z',
+        location_name: 'Rio de Janeiro',
+        overall_aqi_level: 3,
+      }),
+    ]
+
+    const mockedMeasurementsCityPageResponse = [
+      //Lower bound of 00:00
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-06-30T23:29:59Z',
+        pm2_5: 800,
+        pm10: 1200,
+        so2: 1250,
+        no2: 1000,
+        o3: 800,
+        site_name: 'Tijuca',
+      }),
+
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 10,
+        pm10: 20,
+        so2: 100,
+        no2: 40,
+        o3: 50,
+        site_name: 'Tijuca',
+      }),
+      // Upper bound of 00:00
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:30:00Z',
+        pm2_5: 800,
+        pm10: 1200,
+        so2: 1250,
+        no2: 1000,
+        o3: 800,
+        site_name: 'Tijuca',
+      }),
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 20,
+        pm10: 40,
+        so2: 200,
+        no2: 90,
+        o3: 100,
+        site_name: 'AQI 2 Site',
+      }),
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 25,
+        pm10: 50,
+        so2: 350,
+        no2: 120,
+        o3: 130,
+        site_name: 'AQI 3 site',
+      }),
+      createMeasurementsCityPageResponseData({
+        measurement_date: '2024-07-01T00:00:00Z',
+        pm2_5: 50,
+        pm10: 100,
+        so2: 500,
+        no2: 230,
+        o3: 240,
+        site_name: 'AQI 4 site',
+      }),
+
+      //AQI 2 values
+      createMeasurementsCityPageResponseData({
```

Comment:
Could we drop the values so only one of the pollutants in turn is responsible for the value. I.e

at 01:00 only pm10 is 40, everything else is 0
at 02:00 only so2 is 360, everything else is 0.

This way we're also checking that each of the pollutants is able to be the reason for the AQI level

Reply:
Done 💯